client/src/components/CourseReview.tsx

```diff
@@ -27,22 +31,103 @@ export const CourseReview = ({
   showCourse,
   includeTaughtBy = true,
 }: CourseReviewProps) => {
+  const user = useAuth();
+
   showCourse = showCourse ?? false;
 
+  const [kind, setKind] = useState<InteractionKind | undefined>();
+  const [likes, setLikes] = useState(0);
   const [readMore, setReadMore] = useState(false);
+  const [error, setError] = useState('');
 
   const dateStr = format(
     new Date(parseInt(review.timestamp.$date.$numberLong, 10)),
     'PPP'
   );
 
+  useEffect(() => {
+    refreshInteractions();
+  });
+
+  const refreshInteractions = () => {
+    fetchClient
+      .getData<Interaction[]>(
+        `/interactions?course_id=${review.courseId}&user_id=${review.userId}`
+      )
+      .then((interactions) => {
+        setLikes(
+          interactions.filter(
+            (interaction) => interaction.kind === InteractionKind.Like
+          ).length -
+            interactions.filter(
+              (interaction) => interaction.kind === InteractionKind.Dislike
+            ).length
+        );
+        if (user)
+          setKind(
+            interactions.find(
+              (interaction: Interaction) => interaction.referrer === user.id
+            )?.kind
+          );
+      })
```

Comment:
Do you think it'd be better to do this data processing in the backend? Thinking about potential performance problems in the future

Reply:
Yah good call